synapse_invite_checker/rest/contacts.py

```diff
@@ -25,28 +24,20 @@
 from synapse.types import JsonDict
 
 from synapse_invite_checker.config import InviteCheckerConfig
+from synapse_invite_checker.rest.base import (
+    CONTACT_MANAGEMENT_API_PREFIX,
+    invite_checker_pattern,
+)
 from synapse_invite_checker.store import InviteCheckerStore
 from synapse_invite_checker.types import Contact
 
 
-def invite_checker_pattern(path_regex: str, config: InviteCheckerConfig):
-    path = path_regex.removeprefix("/")
-    root = config.api_prefix.removesuffix("/")
-    raw_regex = f"^{root}/{path}"
-
-    # we need to strip the /$, otherwise we can't register for the root of the prefix in a handler...
-    if raw_regex.endswith("/$"):
-        raw_regex = raw_regex.replace("/$", "$")
-
-    return [re.compile(raw_regex)]
-
-
-class InfoResource(RestServlet):
+class ContactManagementInfoResource(RestServlet):
     def __init__(self, config: InviteCheckerConfig, version: str):
         super().__init__()
         self.config = config
         self.version = version
```

Comment:
This should now use the _TMCM_schema_version, imo?

Reply:
Oh, you are passing that as a separate parameter! Should we just make those constants in the config?

Reply:
They are constants, but they are defined in the `InviteChecker` class. I would prefer to have them centrally located, but if you think they should stay in the code for the REST portion of the API, I guess I can move them

Reply:
You can also have them in the invite checker class, but I would like the contacts API to be responsible for knowing, that it is the contacts API. The other files shouldn't need to care, what version gets returned by this endpoint. It splits knowledge, at least in how I parse it. Nitpick though.

Reply:
I think I can see that logic. It would make one additional line to extract when the contact api leaves, which isn't a huge burden but is still something to remember to remove. I'll move those to their respective classes

Reply:
I suppose by the same logic, should probably put the specialized api prefix constants into the same place. That would actually lose a whole file(since that is the only thing in it now)

Reply:
resolved in [a8d782c](https://github.com/famedly/synapse-invite-checker/pull/33/commits/a8d782ce7b5081b7b6b1ce8991d001dd1ecd9bed)

additionals in [76ad966](https://github.com/famedly/synapse-invite-checker/pull/33/commits/76ad966466c81f0bf14d86c9fb01afcd0764bb72)